WordPress/src/main/java/org/wordpress/android/ui/posts/adapters/PostsListAdapter.java

```diff
@@ -729,34 +751,23 @@ protected Boolean doInBackground(Void... nada) {
             }
 
             // Generate the featured image url for each post
-            String imageUrl = null;
             mFeaturedImageUrls.clear();
             for (PostModel post : tmpPosts) {
-                if (post.isLocalDraft()) {
-                    imageUrl = null;
-                } else if (post.getFeaturedImageId() != 0) {
+                String imageUrl = null;
+                if (post.getFeaturedImageId() != 0) {
                     MediaModel media = mMediaStore.getSiteMediaWithId(mSite, post.getFeaturedImageId());
                     if (media != null) {
                         imageUrl = media.getUrl();
                     } else {
-                        // Reset the current `imageUrl` so it doesn't contain the previous post's image
-                        imageUrl = null;
-                    }
-                    // If the imageUrl isn't found it means the featured image info hasn't been added to
-                    // the local media library yet, so add to the list of media IDs to request info for
-                    if (TextUtils.isEmpty(imageUrl)) {
+                        // If the media isn't found it means the featured image info hasn't been added to
+                        // the local media library yet, so add to the list of media IDs to request info for
                         mediaIdsToUpdate.add(post.getFeaturedImageId());
                     }
-                } else if (StringUtils.isNotEmpty(post.getContent())) {
-                    ReaderImageScanner scanner = new ReaderImageScanner(post.getContent(), mSite.isPrivate());
-                    imageUrl = scanner.getLargestImage();
                 } else {
-                    imageUrl = null;
+                    imageUrl = new ReaderImageScanner(post.getContent(), mSite.isPrivate()).getLargestImage();
```

Comment:
Sorry, I missed this in the first pass. Should we use `SiteUtils.isPhotonCapable(mSite)` here?